### PR TITLE
More reliable check for jobs pending indefinitely

### LINKED ISSUE #303: 'ReqNodeNotAvail' is not always a reason of indefinite blocking of a job
In fact, on Kesch this is the reason we get when a job is just pending:

```
ReqNodeNotAvail,  UnavailableNodes:
```

I think here the key is that the `UnavailableNodes` list is empty. Perhaps, we need to have a more robust check before cancelling the job.

### PR DESCRIPTION
**Pull request by teojgo**
* Handle the above case which means that the job is pending.

* Minor codestyle fix.

Fixes #303

### CODE REVIEW

**Comment by teojgo** [comment]
@jenkins-cscs retry daint kesch monch

File: reframe/core/schedulers/slurm.py [hunk 1]
```diff
@@ -266,6 +266,12 @@ def _check_and_cancel(self, reason_descr):
             # no reason details
             reason, reason_details = reason_descr, None
 
+        # Here we handle the case were the UnavailableNodes list is empty,
+        # which actually means that the job is pending
+        if reason == 'ReqNodeNotAvail' and reason_details:
+            if not re.match(r'UnavailableNodes:\S+', reason_details):
```
Comment (vkarak) [inline reframe/core/schedulers/slurm.py]: I think this regex is a bit restrictive, cos you require that whatever comes after `:` has no whitespaces. I'd use instead the following:

```python
if re.match(r'UnavailableNodes:$', reason_details):
    return
```

Comment (vkarak) [reply reframe/core/schedulers/slurm.py]: This whole `if` should go inside the next one, which checks if the reason is of a cancellation type.

Comment (vkarak) [reply reframe/core/schedulers/slurm.py]: Another thing to remove is comment about "inaccurate SLURM documentation" in the definition of the `_cancel_reasons` list.

File: reframe/core/schedulers/slurm.py [hunk 2]
```diff
@@ -267,6 +267,12 @@ def _check_and_cancel(self, reason_descr):
             reason, reason_details = reason_descr, None
 
         if reason in self._cancel_reasons:
+            # Here we handle the case were the UnavailableNodes list is empty,
+            # which actually means that the job is pending
+            if reason == 'ReqNodeNotAvail' and reason_details:
+                if not re.match(r'UnavailableNodes:$', reason_details):
```
Comment (vkarak) [inline reframe/core/schedulers/slurm.py]: The `not` is wrong here. This regex matches only when there is nothing after the `:`.

**Pull request merged**